operate/client/src/modules/utils/flowNodeSelection.test.ts

```diff
@@ -114,3 +117,130 @@ describe('getSelectedRunningInstanceCount', () => {
+      type: 'AD_HOC_SUB_PROCESS_INNER_INSTANCE',
+    });
+
+    const mockChild = createMockFlowNodeInstance({
+      id: 'child-1',
+      flowNodeId: 'task-1',
+      type: 'SERVICE_TASK',
+    });
+
+    // Mock that children are already loaded
+    const getVisibleChildNodesSpy = vi
+      .spyOn(flowNodeInstanceStore, 'getVisibleChildNodes')
+      .mockReturnValue([mockChild]);
+
+    selectAdHocSubProcessInnerInstance(mockRootNode, mockInnerInstance);
+
+    expect(getVisibleChildNodesSpy).toHaveBeenCalledWith(mockInnerInstance);
+    expect(flowNodeSelectionStore.state.selection).toEqual({
+      flowNodeId: 'task-1',
+      flowNodeInstanceId: 'child-1',
+    });
+
+    getVisibleChildNodesSpy.mockRestore();
+  });
+
+  it('should handle case when there are no children available', () => {
+    const mockRootNode = {flowNodeId: 'root', flowNodeInstanceId: 'root-1'};
+    const mockInnerInstance = createMockFlowNodeInstance({
+      id: 'inner-1',
+      flowNodeId: 'adhoc-subprocess',
+      type: 'AD_HOC_SUB_PROCESS_INNER_INSTANCE',
+    });
+
+    // Mock that no children are loaded initially
+    const getVisibleChildNodesSpy = vi
+      .spyOn(flowNodeInstanceStore, 'getVisibleChildNodes')
+      .mockReturnValue([]);
+
+    selectAdHocSubProcessInnerInstance(mockRootNode, mockInnerInstance);
+
+    expect(getVisibleChildNodesSpy).toHaveBeenCalledWith(mockInnerInstance);
+    expect(flowNodeSelectionStore.state.selection).toBeNull();
+
+    getVisibleChildNodesSpy.mockRestore();
+  });
+
+  it('should wait for lazy-loaded children and select first child when available', () => {
+    const mockRootNode = {flowNodeId: 'root', flowNodeInstanceId: 'root-1'};
+    const mockInnerInstance = createMockFlowNodeInstance({
+      id: 'inner-1',
+      flowNodeId: 'adhoc-subprocess',
+      type: 'AD_HOC_SUB_PROCESS_INNER_INSTANCE',
+    });
+
+    const getVisibleChildNodesSpy = vi
+      .spyOn(flowNodeInstanceStore, 'getVisibleChildNodes')
+      .mockReturnValue([]); // Always return empty for this test
+
+    selectAdHocSubProcessInnerInstance(mockRootNode, mockInnerInstance);
+
+    expect(flowNodeSelectionStore.state.selection).toBeNull();
+    expect(getVisibleChildNodesSpy).toHaveBeenCalledWith(mockInnerInstance);
+
+    getVisibleChildNodesSpy.mockRestore();
+  });
+
+  it('should handle multiple children and select only the first one', () => {
+    const mockRootNode = {flowNodeId: 'root', flowNodeInstanceId: 'root-1'};
+    const mockInnerInstance = createMockFlowNodeInstance({
+      id: 'inner-1',
+      flowNodeId: 'adhoc-subprocess',
+      type: 'AD_HOC_SUB_PROCESS_INNER_INSTANCE',
+    });
+
+    const mockFirstChild = createMockFlowNodeInstance({
+      id: 'child-1',
+      flowNodeId: 'task-1',
+      type: 'SERVICE_TASK',
+    });
+
+    const mockSecondChild = createMockFlowNodeInstance({
+      id: 'child-2',
+      flowNodeId: 'task-2',
+      type: 'USER_TASK',
+    });
+
+    // Mock multiple children
```

Comment:
All the comments on this file are unnecessary.

Comments should be used in cases to explain why something was done, not what's being done